api/deployments/component_handler.go

```diff
@@ -129,46 +132,130 @@ func GetComponentStateFromSpec(
 		if err != nil {
 			return nil, err
 		}
-		podNames = getPodNames(pods)
-		environmentVariables = getRadixEnvironmentVariables(pods)
-		replicaSummaryList = getReplicaSummaryList(pods)
+		componentPodNames, componentPods = slicePodNamesAndPodsFromMap(componentPodMap)
+		environmentVariables = getRadixEnvironmentVariables(componentPods)
+		replicaSummaryList = getReplicaSummaryList(*componentPods)
 
+		scheduledJobs, err := getComponentJobsByNamespace(client, envNs, component.GetName()) //scheduledJobs
+		if err != nil {
+			return nil, err
+		}
+		scheduledJobSummaryList = getScheduledJobSummaryList(scheduledJobs, podsOfScheduledJobsMap)
 		status, err = getStatusOfActiveDeployment(component,
-			deploymentStatus, environmentConfig, pods)
+			deploymentStatus, environmentConfig, *componentPods)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	return deploymentModels.NewComponentBuilder().
+	componentBuilder := deploymentModels.NewComponentBuilder()
+	if jobComponent, ok := component.(*v1.RadixDeployJobComponent); ok {
+		componentBuilder.WithSchedulerPort(jobComponent.SchedulerPort)
+		if jobComponent.Payload != nil {
+			componentBuilder.WithScheduledJobPayloadPath(jobComponent.Payload.Path)
+		}
+	}
+	return componentBuilder.
 		WithComponent(component).
 		WithStatus(status).
-		WithPodNames(podNames).
+		WithPodNames(*componentPodNames).
 		WithReplicaSummaryList(replicaSummaryList).
+		WithScheduledJobSummaryList(scheduledJobSummaryList).
 		WithRadixEnvironmentVariables(environmentVariables).
 		BuildComponent(), nil
 }
 
-func getComponentPodsByNamespace(client kubernetes.Interface, envNs, componentName string) ([]corev1.Pod, error) {
+func getScheduledJobSummaryList(jobs []batchv1.Job, pods *map[string][]corev1.Pod) []deploymentModels.ScheduledJobSummary {
+	var summaries []deploymentModels.ScheduledJobSummary
+	for _, job := range jobs {
+		creationTimestamp := job.GetCreationTimestamp()
+		summary := deploymentModels.ScheduledJobSummary{
+			Name:    job.Name,
+			Created: utils.FormatTimestamp(creationTimestamp.Time),
+			Started: utils.FormatTime(job.Status.StartTime),
+			Ended:   utils.FormatTime(job.Status.CompletionTime),
+			Status:  models.GetStatusFromJobStatus(job.Status).String(),
+		}
+		if jobPods, ok := (*pods)[job.Name]; ok {
+			summary.ReplicaList = getReplicaSummariesForPods(jobPods)
+		}
+		summaries = append(summaries, summary)
+	}
+
+	// Sort job-summaries descending
+	sort.Slice(summaries, func(i, j int) bool {
+		return utils.IsBefore(&summaries[j], &summaries[i])
+	})
+	return summaries
+}
+
+func getReplicaSummariesForPods(jobPods []corev1.Pod) []deploymentModels.ReplicaSummary {
+	var replicaSummaries []deploymentModels.ReplicaSummary
+	for _, pod := range jobPods {
+		replicaSummaries = append(replicaSummaries, deploymentModels.GetReplicaSummary(pod))
+	}
+	return replicaSummaries
+}
+
+func slicePodNamesAndPodsFromMap(podMap *map[string]corev1.Pod) (*[]string, *[]corev1.Pod) {
```

Comment:
I don't think it is neccessary to use pointers to slices. A slice is a struct with a pointer to an array.

Also, maps are always reference types. No need to use pointers to map as parameters.

```
func changeMap(m map[string]string) {
	m["b"] = "bb"
}

func main() {
	var m map[string]string = make(map[string]string)
	m["a"] = "aa"
	changeMap(m)
	fmt.Println(m)
	// prints map[a:aa b:bb]
}
```

Reply:
fixed